### ISSUE TITLE
Crash when using spawn function on Android

### ISSUE THREAD

**Issue by aku-ato** [frida/frida-go#13]
Description:

I am encountering a crash issue while using the spawn function in the frida-go library. Whenever I try to use the spawn function, the library crashes and I am unable to proceed with my task.

Steps to reproduce the issue:

    Import the frida-go library into a project
    Use the spawn function to spawn a process
    Observe that the library crashes 

Expected behavior:

I expect the spawn function to successfully spawn the process without any crashes.

Actual behavior:

The frida-go library crashes when the spawn function is used, making it impossible to continue

Additional information:

    I am using the latest version of frida (16.0.10) the frida-go (v0.6.1) library.
    I m testing on Android 10 - Mi A2  
    The crash occurs consistently.

Please let me know if there is any additional information that would be helpful in diagnosing and resolving this issue. Thank you.

```go
package main

import (
	f "github.com/frida/frida-go/frida"
)

func main() {
	dm := f.NewDeviceManager()

	_, err := dm.EnumerateDevices()
	DoneOrDie(err)

	d, err := dm.FindDeviceByID("dd1e7d")
	DoneOrDie(err)

	spawn, err := d.Spawn("com.myapp", nil)
	DoneOrDie(err)

	session, err := d.Attach(spawn, nil)
	DoneOrDie(err)

	script, err := session.CreateScript("console.log('Hello, world!');")
	DoneOrDie(err)

	script.On("message", func(message *f.Message, data []byte) {
		println(message)
	})

	err = script.Load()
	DoneOrDie(err)

	<-make(chan bool)
}

func DoneOrDie(err error) {
	if err != nil {
		panic(err)
	}
}
```

```
goroutine 1 [syscall]:
runtime.cgocall(0x1023a4b4c, 0x1400005bdc8)
        /Users/paolo/sdk/go1.19.4/src/runtime/cgocall.go:158 +0x54 fp=0x1400005bd90 sp=0x1400005bd50 pc=0x1022da274
github.com/frida/frida-go/frida._Cfunc_g_object_unref(0x0)
        _cgo_gotypes.go:3452 +0x34 fp=0x1400005bdc0 sp=0x1400005bd90 pc=0x10239d234
github.com/frida/frida-go/frida.unrefGObj.func1(0x63e400005eac160?)
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/cleanups.go:30 +0x40 fp=0x1400005be00 sp=0x1400005bdc0 pc=0x10239e1b0
github.com/frida/frida-go/frida.unrefGObj(0x105eb0600?)
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/cleanups.go:30 +0x1c fp=0x1400005be20 sp=0x1400005be00 pc=0x10239e14c
github.com/frida/frida-go/frida.clean(0x0?, {0x1028cd1b6?, 0x140000aa001?})
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/cleanups.go:36 +0x58 fp=0x1400005be60 sp=0x1400005be20 pc=0x10239e228
github.com/frida/frida-go/frida.(*Device).Spawn.func3()
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/device.go:346 +0x34 fp=0x1400005be90 sp=0x1400005be60 pc=0x10239f744
github.com/frida/frida-go/frida.(*Device).Spawn(0x140000ae290, {0x1028ccb33?, 0x1023a81d0?}, 0x0?)
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/device.go:354 +0x138 fp=0x1400005bf20 sp=0x1400005be90 pc=0x10239f428
main.main()
...
```

**Comment by NSEcho** [comment]
Hi, can you show me the output when you add `frida.PatchAndroidSELinux()` line before `dm := f.NewDeviceManager()`.

**Comment by aku-ato** [comment]
```

GOROOT=/opt/homebrew/Cellar/go/1.19.5/libexec #gosetup
GOPATH=/Users/paolo/go #gosetup
/opt/homebrew/Cellar/go/1.19.5/libexec/bin/go build -o /private/var/folders/fz/mhjt90cx1q3g8d2zg1lhwjd40000gn/T/GoLand/___11go_build_test-frida-go test-frida-go #gosetup
/private/var/folders/fz/mhjt90cx1q3g8d2zg1lhwjd40000gn/T/GoLand/___11go_build_test-frida-go
Device: Mi A2
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x102a7df90]

runtime stack:
runtime.throw({0x102b380f1?, 0x0?})
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/panic.go:1047 +0x40 fp=0x16d8cb150 sp=0x16d8cb120 pc=0x102569f90
runtime.sigpanic()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/signal_unix.go:819 +0x1e4 fp=0x16d8cb180 sp=0x16d8cb150 pc=0x10257ebb4

goroutine 1 [syscall]:
runtime.cgocall(0x10260514c, 0x1400005bda8)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/cgocall.go:158 +0x54 fp=0x1400005bd70 sp=0x1400005bd30 pc=0x10253a274
github.com/frida/frida-go/frida._Cfunc_g_object_unref(0x0)
        _cgo_gotypes.go:3452 +0x34 fp=0x1400005bda0 sp=0x1400005bd70 pc=0x1025fd284
github.com/frida/frida-go/frida.unrefGObj.func1(0x3260e3006110300?)
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/cleanups.go:30 +0x40 fp=0x1400005bde0 sp=0x1400005bda0 pc=0x1025fe200
github.com/frida/frida-go/frida.unrefGObj(0x106114800?)
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/cleanups.go:30 +0x1c fp=0x1400005be00 sp=0x1400005bde0 pc=0x1025fe19c
github.com/frida/frida-go/frida.clean(0x0?, {0x102b311cd?, 0x14000010060?})
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/cleanups.go:36 +0x58 fp=0x1400005be40 sp=0x1400005be00 pc=0x1025fe278
github.com/frida/frida-go/frida.(*Device).Spawn.func3()
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/device.go:346 +0x34 fp=0x1400005be70 sp=0x1400005be40 pc=0x1025ff794
github.com/frida/frida-go/frida.(*Device).Spawn(0x140000200f0, {0x102b32b01?, 0xb?}, 0x10610f6c0?)
        /Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/device.go:357 +0x180 fp=0x1400005bf00 sp=0x1400005be70 pc=0x1025ff4c0
main.main()
        /Users/paolo/Projects/medusa/code/reverse/test-frida-go/main.go:50 +0x250 fp=0x1400005bf70 sp=0x1400005bf00 pc=0x1026048d0
runtime.main()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:250 +0x24c fp=0x1400005bfd0 sp=0x1400005bf70 pc=0x10256c50c
runtime.goexit()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x1400005bfd0 sp=0x1400005bfd0 pc=0x102596b44

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000048fa0 sp=0x14000048f80 pc=0x10256c8d4
runtime.goparkunlock(...)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:369
runtime.forcegchelper()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:302 +0xac fp=0x14000048fd0 sp=0x14000048fa0 pc=0x10256c76c
runtime.goexit()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000048fd0 sp=0x14000048fd0 pc=0x102596b44
created by runtime.init.6
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:290 +0x24

goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000049770 sp=0x14000049750 pc=0x10256c8d4
runtime.goparkunlock(...)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:369
runtime.bgsweep(0x0?)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mgcsweep.go:278 +0xa4 fp=0x140000497b0 sp=0x14000049770 pc=0x10255a424
runtime.gcenable.func1()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mgc.go:178 +0x28 fp=0x140000497d0 sp=0x140000497b0 pc=0x10254e9e8
runtime.goexit()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x140000497d0 sp=0x140000497d0 pc=0x102596b44
created by runtime.gcenable
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mgc.go:178 +0x74

goroutine 4 [GC scavenge wait]:
runtime.gopark(0x14000070000?, 0x102b59ec0?, 0x1?, 0x0?, 0x0?)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000049f50 sp=0x14000049f30 pc=0x10256c8d4
runtime.goparkunlock(...)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:369
runtime.(*scavengerState).park(0x1061dbdc0)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mgcscavenge.go:389 +0x5c fp=0x14000049f80 sp=0x14000049f50 pc=0x1025583dc
runtime.bgscavenge(0x0?)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mgcscavenge.go:617 +0x44 fp=0x14000049fb0 sp=0x14000049f80 pc=0x102558984
runtime.gcenable.func2()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mgc.go:179 +0x28 fp=0x14000049fd0 sp=0x14000049fb0 pc=0x10254e988
runtime.goexit()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000049fd0 sp=0x14000049fd0 pc=0x102596b44
created by runtime.gcenable
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mgc.go:179 +0xb8

goroutine 18 [finalizer wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000044580 sp=0x14000044560 pc=0x10256c8d4
runtime.goparkunlock(...)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/proc.go:369
runtime.runfinq()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mfinal.go:180 +0x120 fp=0x140000447d0 sp=0x14000044580 pc=0x10254dc10
runtime.goexit()
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x140000447d0 sp=0x140000447d0 pc=0x102596b44
created by runtime.createfing
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/mfinal.go:157 +0x84

Process finished with the exit code 2
```

**Comment by aku-ato** [comment]
fixed my code with this:

```go
	opt := f.NewSpawnOptions()

	spawn, err := d.Spawn("com.myapp", opt)
	DoneOrDie(err)
```

the issue seem to be here [/Users/paolo/go/pkg/mod/github.com/frida/frida-go@v0.6.1/frida/device.go](https://github.com/frida/frida-go/blob/ac54cbc9f77c2b87b3b9ff0b69e604cee596a42a/frida/device.go#L346) : 

> defer clean(unsafe.Pointer(opt), unrefFrida)

`nil` case the issue

**Comment by NSEcho** [comment]
Yeah, I have come to the same conclusion, will push the fix soon

**Comment by NSEcho** [comment]
Can you just confirm that it is working correctly with the original code when using frida-go `v0.6.2`

**Comment by aku-ato** [comment]
Yes it's fixed in `v0.6.2`. 

Thank you.
